renderer/reducers/account.js

```diff
@@ -0,0 +1,266 @@
+
+export const CHANGE_PASSWORD_DIALOG_ID = 'CHANGE_PASSWORD_DIALOG'
+export const PASSWORD_PROMPT_DIALOG_ID = 'PASSWORD_PROMPT_DIALOG'
+export const PASSWORD_SET_DIALOG_ID = 'PASSWORD_SET_DIALOG_ID'
+
+// ------------------------------------
+// Actions
+// ------------------------------------
+
+/**
+ * initAccounts - Fetch the current account info from the database and save into the store.
+ * Should be called once when the app first loads.
+ *
+ * @returns {Function} Thunk
+ */
+export const initAccount = () => async (dispatch, getState) => {
+  dispatch({ type: INIT_ACCOUNT })
+  try {
+    // Auto login user if password feature is disabled.
+    const isAccountPasswordEnabled = accountSelectors.isAccountPasswordEnabled(getState())
+    if (!isAccountPasswordEnabled) {
+      dispatch({ type: LOGIN_SUCCESS })
+    }
+    dispatch({ type: INIT_ACCOUNT_SUCCESS })
+  } catch (error) {
+    dispatch({ type: INIT_ACCOUNT_FAILURE, error: error.message })
+  }
+}
+
+/**
+ * setPassword - Updates wallet password.
+ *
+ * @param {string} password new password
+ * @returns {Function} Thunk
+ */
+const setPassword = password => async dispatch => {
+  dispatch(send('setPassword', { password: await sha256digest(password) }))
+}
+
+/**
+ * changePassword - Changes existing password.
+ *
+ * @param {object} params password params
+ * @param {string} params.newPassword new password
+ * @param {string} params.oldPassword old password
+ * @returns {Function} Thunk
+ */
+export const changePassword = ({ newPassword, oldPassword }) => async dispatch => {
+  try {
+    const intl = getIntl()
+    await dispatch(requirePassword(oldPassword))
+    await dispatch(setPassword(newPassword))
+    dispatch(closeDialog(CHANGE_PASSWORD_DIALOG_ID))
+    dispatch(showNotification(intl.formatMessage(messages.account_password_updated)))
+  } catch (error) {
+    dispatch({ type: LOGIN_FAILURE, error: error.message })
+  }
+}
+
+/**
+ * enablePassword - Enables app-wide password protection.
+ *
+ * @param {string} password to be used further on.
+ * @returns {Function} Thunk
+ */
+export const enablePassword = ({ password }) => async dispatch => {
+  try {
+    const intl = getIntl()
+    dispatch(setPassword(password))
+    dispatch(
+      saveConfigOverrides({
+        password: {
+          active: true,
+        },
+      })
+    )
+    dispatch(closeDialog(PASSWORD_SET_DIALOG_ID))
+    dispatch(showNotification(intl.formatMessage(messages.account_password_enabled)))
+  } catch (error) {
+    dispatch({ type: LOGIN_FAILURE, error: error.message })
+  }
+}
+
+/**
+ * enablePassword - Disables app-wide password protection.
+ *
+ * @param {string} password current password.
+ * @returns {Function} Thunk
+ */
+export const disablePassword = ({ password }) => async dispatch => {
+  try {
+    const intl = getIntl()
+    await dispatch(requirePassword(password))
+    dispatch(send('deletePassword'))
+    dispatch(
+      saveConfigOverrides({
```

Comment:
Can also use `putConfig`

Reply:
oh nice. this is exactly why i was looking for but forgot `putConfig` exists

Reply:
this was removed all together